kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiSyncThriftClient.scala

```diff
@@ -161,8 +164,10 @@ class KyuubiSyncThriftClient private (
       case e: Exception =>
         throw KyuubiSQLException("Error while cleaning up the engine resources", e)
     } finally {
-      Option(engineAliveThreadPool).foreach(_.shutdown())
-      if (_aliveProbeSessionHandle != null && !remoteEngineBroken) {
+      Option(engineAliveThreadPool).foreach { pool =>
+        ThreadUtils.shutdown(pool, Duration(engineAliveProbeInterval, TimeUnit.MILLISECONDS))
```

Comment:
wait the current task finished to prevent out of sequence issue.

Reply:
engineAliveProbeInterval -> connectionTimeout ?

Reply:
do you mean rename the `engineAliveProbeInterval` variable?

Reply:
If the probe request is in flight, the max flight time should not be `engineAliveProbeInterval`. We should use a more accurate time.

Reply:
engineAliveProbeInterval is also the requestTimeout of the alive probe thrift client.

```
          createTProtocol(user, passwd, host, port, aliveProbeInterval, loginTimeout),
```

Reply:
ah I see, then it's ok